### PR TITLE
Add dependency rake task

### PR DESCRIPTION
**Pull request by cashmonger**

### CODE REVIEW

File: lib/tasks/dependencies.rake [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        if version.user_id != nil
+          user_names << User.select(:name).find(version.user_id).name 
+        end 
+      end
+      return user_names.uniq! 
+    end 
+
+    f = File.new("lib/dependencies.csv", "w") 
+    
+    f.puts "operation type category/name, [libraries cited, methods cited], number of libraries cited, #libraries within category, #libraries outside of category, developers who worked on code"
```
Comment (dvnstrcklnd) [inline lib/tasks/dependencies.rake]: Commas in a csv file are reserved for delimeters.

Comment (bjkeller) [reply lib/tasks/dependencies.rake]: I think the tabular summary data can go to CSV and the relational data should go to a JSON file.

Comment (dvnstrcklnd) [reply lib/tasks/dependencies.rake]: Maybe, but it's probably destined for a denormalized pandas dataframe anyway.

**Comment by bjkeller** [comment]
@cashmonger Can you take a look at the changes I've made and fill out the rest?  I left some TODOs.  Otherwise, let's work through this together tomorrow.

**Pull request merged**